app/src/main/java/com/eveningoutpost/dexdrip/NFCReaderX.java

```diff
@@ -277,7 +277,11 @@ public static boolean HandleGoodReading(final String tagId, byte[] data1, final
     
     // returns true if checksum passed.
     public static boolean HandleGoodReading(final String tagId, byte[] data1, final long CaptureDateTime, final boolean allowUpload, byte []patchUid,  byte []patchInfo, boolean decripted_data ) {
-        Log.e(TAG, "HandleGoodReading called");
+        Log.e(TAG, "HandleGoodReading called dat1 len = " + data1.length);
+        if(data1.length > 0x158) {
+            // It seems that some times we read a buffer that is bigger than 0x158, but we should only use the first 0x158 bytes.
+            data1 = java.util.Arrays.copyOfRange(data1, 0, 0x158);
```

Comment:
Is this save for all Libre NFC readings? Maybe we should declare a constant for it and use this constant to iterate over the array rather than shrinking it.

Reply:
Right now it is the biggest libre buffer that we are aware of, and many places it is used to check for a legal buffer.
I'll change it to a constant later, and you will see all it's hidden usages.